### ISSUE TITLE
Retirement of Office 365 connectors within Microsoft Teams

### ISSUE THREAD

**Issue by ryanbuening** [AntoineGa/Seq.App.Teams#47]
It appears that Microsoft is [retiring Office 365 connectors within Microsoft Teams](https://devblogs.microsoft.com/microsoft365dev/retirement-of-office-365-connectors-within-microsoft-teams/).

I was wondering if this app will continue to work with the new Workflows app they are pushing users to? I noticed that there is a Power Automate template for "Post card to channel in Microsoft Teams when webhook request is received", however, I'm unsure if Seq.App.Teams would need updated to accommodate the Microsoft changes.

Or maybe there is a different/better way to handle pushing alerts to Teams from Seq that I'm unaware of? @nblumhardt

**Comment by nblumhardt** [comment]
Hi Ryan,

I checked out the linked post, I can't make a lot of sense of it. Incoming webhook connectors are so widely-used it'd be really, really strange for Microsoft to retire them with three months notice.

The docs for creating webhooks don't mention any kind of deprecation:

https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook?tabs=newteams%2Cdotnet#create-an-incoming-webhook

I'm unfortunately not the best person to answer (very little experience with Teams) so someone paying closer attention may be able to clarify this :) Cheers!

**Comment by ryanbuening** [comment]
> Incoming webhook connectors are so widely-used it'd be really, really strange for Microsoft to retire them with three months notice.

Yeah, people don't seem too happy about it in the [comments](https://devblogs.microsoft.com/microsoft365dev/retirement-of-office-365-connectors-within-microsoft-teams/#comments). 🫤

![image](https://github.com/AntoineGa/Seq.App.Teams/assets/812974/03804bf9-6cbd-4888-9872-0ec2a750dde3)

Here is what the Seq alert looks like now with the `Action Required` message:

![image](https://github.com/AntoineGa/Seq.App.Teams/assets/812974/f8b3b4b6-af55-49a4-9ab5-7b37118bb60e)

**Comment by nblumhardt** [comment]
Thanks for posting the screenshot. What happens if you click "Set up workflow"? It sounds like the webhook functionality is just being moved there, in which case this app would still be the way to trigger it. Keen to help but not sure where to start at this point :-)

**Comment by ryanbuening** [comment]
@nblumhardt it looks like converting to [Seq.App.Teams.AdaptiveCard](https://github.com/MaceWindu/Seq.App.Teams.AdaptiveCard) (and using Workflows) might be the answer here, so I think we're good. Thanks!

Relevant issue in `Seq.App.Teams.AdaptiveCard` [here](https://github.com/MaceWindu/Seq.App.Teams.AdaptiveCard/issues/15).

**Comment by bbenouarets** [comment]
The deadline was actually extended. However, the URL of the webhooks must be updated again by December 31, 2024.

If other system administrators are having difficulties locating the affected teams:
I have written a small tool in Golang that uses the Graph API to output the affected teams.

[Teams Webhook Finder](https://github.com/SecNex/teams-webhook-finder)

This has helped us enormously, as Microsoft does not offer its own solution for reading the affected channels and teams. We have over 350 teams in our company, which we would otherwise have had to search through manually.

I hope I could help someone here with this.

Independently of my day job, I would like to create an alternative to the native “incoming webhooks”. Maybe someone will benefit from it.

**Comment by martin1cerny** [comment]
I'm just trying to upgrade by re-pointing to the new Power Automate Teams webhook, but it seems like it expects a different message schema. Is there anything I can do to change this?

**Comment by martin1cerny** [comment]
I ended up transforming the content to an adaptive card on-the-fly. Just in case this helps anyone as an inspiration.

![Image](https://github.com/user-attachments/assets/8d358088-169c-4922-bf26-beaf558dc611)

**Comment by nblumhardt** [comment]
Nice solution! FWIW `Seq.App.Teams.AdaptiveCard` is a fork that uses the Adaptive Card API natively. HTH!

**Comment by martin1cerny** [comment]
Thanks, I'm aware of this fork, but it didn't work with our version of Seq.
